src/integrant/core.cljc

```diff
@@ -13,10 +13,20 @@
 (defrecord Ref    [key] RefLike (ref-key [_] key))
 (defrecord RefSet [key] RefLike (ref-key [_] key))
 
+(def ^{:private true} composite-key? vector?)
+
+(defn- valid-composite-key?
+  "Returns true if the key is a vector of namespaced keywords."
+  [key]
+  (and (composite-key? key)
+       (every? keyword? key)
+       (every? namespace key)))
+
 (defn valid-config-key?
```

Comment:
Let's make it something like:

```clojure
(defn- namespaced-keyword? [key]
  (and (keyword? key) (namespace key)))

(defn- composite-key? [key]
  (and (vector? key) (every? namespaced-keyword? key)))

(defn valid-config-key?
  "Returns true if the key is a keyword or a vector of namespaced keywords."
  (or (keyword? key) (composite-key? key)))
```

I think in another commit I'll remove non-namespaced keywords, so a key is only valid if it's a namespaced keyword, or vector of namespaced keywords.